WalletWasabi/Wallets/Wallet.cs

```diff
@@ -109,22 +114,23 @@ public class Wallet : BackgroundService
 		public event EventHandler<Block> NewBlockProcessed;
 
 		public WalletState State { get; private set; }
-		public BitcoinStore BitcoinStore { get; }
+		public string DataDir { get; }
+		public BitcoinStore BitcoinStore { get; private set; }
 		public KeyManager KeyManager { get; }
-		public WasabiSynchronizer Synchronizer { get; }
-		public CoinJoinClient ChaumianClient { get; }
-		public NodesGroup Nodes { get; }
+		public WasabiSynchronizer Synchronizer { get; private set; }
+		public CoinJoinClient ChaumianClient { get; private set; }
+		public NodesGroup Nodes { get; private set; }
 		public string BlocksFolderPath { get; }
-		public ServiceConfiguration ServiceConfiguration { get; }
+		public ServiceConfiguration ServiceConfiguration { get; private set; }
```

Comment:
The `WalletManager` contains exactly the same properties. Why not pass the `WalletManager` in the constructor?

Reply:
Two reasons: 

1. Many tests were designed to Wallet (WalletService) live on its own without knowing there is a WalletManager.
2. In the last period, we are decoupling things like global because it was leading to tether classes which not belong to each other. So I prefer to add the logical components one-by-one to every class, so later modularization becomes easier.